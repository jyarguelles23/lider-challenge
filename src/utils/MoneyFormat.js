export const formatAmount = (amount) =>
	amount.toString().replace(/(\d)(?=(\d{3})+(?:\.\d+)?$)/g, '$1.');
