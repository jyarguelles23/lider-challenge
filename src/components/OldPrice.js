import React from "react";
import { formatAmount } from "../utils/MoneyFormat";

const OldPrice = (oldPrices) => {
  const { oldPrice } = oldPrices;
  return (
    <div className="product-old-price">
      {oldPrice === 0 ? null : `$ ${formatAmount(oldPrice)}`}
    </div>
  );
};

export default OldPrice;
