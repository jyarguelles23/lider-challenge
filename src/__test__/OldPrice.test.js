import OldPrice from "../components/OldPrice";
import { formatAmount } from "../utils/MoneyFormat";
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";

let tree = ReactTestRenderer;
const oldPrice = 500.000;

describe("<OldPrice/>", () => {
  beforeEach(() => {
    tree = renderer.create(<OldPrice oldPrice={oldPrice} />);
  });

  it("should render without crashing", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render the oldprice formatted", () => {
    const price = tree.root;
    const oldPriceFormatted = formatAmount(oldPrice);
    expect(
      price.findByProps({ className: "product-old-price" }).children
    ).toEqual([`$ ${oldPriceFormatted}`]);
  });
});
