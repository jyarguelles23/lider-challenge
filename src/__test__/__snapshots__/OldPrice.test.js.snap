// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OldPrice/> should render without crashing 1`] = `
<div
  className="product-old-price"
>
  $ 500
</div>
`;
